### PR TITLE
Fix: use standard ASPNET Core namespace for extensions

### PR DESCRIPTION
**Pull request by seangwright**
# Motivation

All DI related extension methods (extensions on `IServiceCollection`) should be in the same namespace - `Microsoft.Extensions.DependencyInjection`. This makes it easier for developers to find these extensions when they are configuring dependencies.

This PR updates the extensions in this library to follow that convention.

## Checklist

- [X] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [X] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

## How to test

n/a

### CODE REVIEW

**Comment by seangwright** [comment]
This PR requires #5 to be merged first, because the repository contains an extra `.sln` file which breaks the CI build.

**Comment by bkapustik** [comment]
> This PR requires #5 to be merged first, because the repository contains an extra `.sln` file which breaks the CI build.

This created unused using in Dancing goat Program.cs. I have pushed a fix. Otherwise this is fine with me.

**Review by bkapustik — APPROVED** [review]

File: .azuredevops/pipelines/build-and-release.yml [hunk 1]
```diff
@@ -44,10 +44,10 @@ stages:
         variables:
           - group: Code Sign KV Auth
 
-          - name: Configuration
-            value: Release
+           - name: ProjectFolder
```
Comment (michalJakubis) [inline .azuredevops/pipelines/build-and-release.yml:47]: This change in indentation has broke the azure pipeline.

Comment (seangwright) [reply .azuredevops/pipelines/build-and-release.yml:47]: This is fixed in #5

**Pull request merged**